receiver/netflowreceiver/receiver.go

```diff
@@ -5,25 +5,165 @@ package netflowreceiver // import "github.com/open-telemetry/opentelemetry-colle
-	config      *Config
-	logConsumer consumer.Logs
+	config      Config
 	logger      *zap.Logger
-	// listeners   []*Listener
+	udpReceiver *utils.UDPReceiver
+	logConsumer consumer.Logs
+}
+
+func newNetflowLogsReceiver(params receiver.Settings, cfg Config, consumer consumer.Logs) (receiver.Logs, error) {
+	// UDP receiver configuration
+	udpCfg := &utils.UDPReceiverConfig{
+		Sockets:   cfg.Sockets,
+		Workers:   cfg.Workers,
+		QueueSize: cfg.QueueSize,
+		Blocking:  false,
+		ReceiverCallback: &dropHandler{
+			logger: params.Logger,
+		},
+	}
+	udpReceiver, err := utils.NewUDPReceiver(udpCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	nr := &netflowReceiver{
+		logger:      params.Logger,
+		config:      cfg,
+		logConsumer: consumer,
+		udpReceiver: udpReceiver,
+	}
+
+	return nr, nil
 }
 
-func (nr *netflowReceiver) Start(_ context.Context, _ component.Host) error {
+func (nr *netflowReceiver) Start(ctx context.Context, host component.Host) error {
+	nr.logger.Info("NetFlow receiver is starting...")
+
+	// The function that will decode packets
+	decodeFunc, err := nr.buildDecodeFunc()
+	if err != nil {
+		return err
+	}
+
+	nr.logger.Info("Start listening for NetFlow on UDP", zap.Any("config", nr.config))
+	if err := nr.udpReceiver.Start(nr.config.Hostname, nr.config.Port, decodeFunc); err != nil {
+		return err
+	}
+
+	// This runs until the receiver is stoppped, consuming from an error channel
+	go nr.handleErrors()
+
 	return nil
 }
 
-func (nr *netflowReceiver) Shutdown(_ context.Context) error {
+func (nr *netflowReceiver) Shutdown(context.Context) error {
+	nr.logger.Info("NetFlow receiver is shutting down...")
+	if nr.udpReceiver == nil {
+		return nil
+	}
+	err := nr.udpReceiver.Stop()
+	if err != nil {
+		nr.logger.Warn("Error stopping UDP receiver", zap.Error(err))
+	}
 	return nil
 }
+
+// buildDecodeFunc creates a decode function based on the scheme
+// This is the fuction that will be invoked for every netflow packet received
+// The function depends on the type of schema (netflow, sflow, flow)
+func (nr *netflowReceiver) buildDecodeFunc() (utils.DecoderFunc, error) {
+	// Eventually this can be used to configure mappings
+	cfgProducer := &protoproducer.ProducerConfig{}
+	cfgm, err := cfgProducer.Compile() // converts configuration into a format that can be used by a protobuf producer
+	if err != nil {
+		return nil, err
+	}
+	// We use a goflow2 proto producer to produce messages using protobuf format
+	protoProducer, err := protoproducer.CreateProtoProducer(cfgm, protoproducer.CreateSamplingSystem)
+	if err != nil {
+		return nil, err
+	}
+
+	// the otel log producer converts those messages into OpenTelemetry logs
+	// it is a wrapper around the protobuf producer
+	otelLogsProducer := newOtelLogsProducer(protoProducer, nr.logConsumer)
+
+	cfgPipe := &utils.PipeConfig{
+		Producer: otelLogsProducer,
+	}
+
+	var decodeFunc utils.DecoderFunc
+	var p utils.FlowPipe
+	switch nr.config.Scheme {
```

Comment:
Could we explain in the readme what protocol support users can expect when configuring one of these three schemes? I had to dig into the GoFlow2 source to understand what is happening here.